src/PotentialSimulation/PotentialSimulationSetups/PotentialSimulationSetupRBCartesian3D.jl

```diff
@@ -185,6 +188,20 @@ function PotentialSimulationSetupRB(det::SolidStateDetector{T}, grid::CartesianG
                 fill_ρ_and_ϵ!(ϵ, ρ_tmp, q_eff_fix_tmp, Cartesian, mpz, mpy, mpx, use_nthreads, passive)
             end
         end
+        if depletion_handling
+            for iz in axes(ϵ, 3)
+                for iy in axes(ϵ, 2)
+                    for ix in axes(ϵ, 1)
+                        pt::CartesianPoint{T} = CartesianPoint{T}(mpx[ix], mpy[iy], mpz[iz])
+                        ig = find_closest_gridpoint(pt, point_types.grid)
+                        if is_undepleted_point_type(point_types.data[ig...]) 

```

Comment:
There is a function `getindex(point_types::PointTypes, pt::AbstractCoordinatePoint)` that does exactly this [on branch v0.7](https://github.com/JuliaPhysics/SolidStateDetectors.jl/blob/v0.7/src/Types/point_types.jl#L58) (`point_types.data[find_closest_gridpoint(pt, point_types.grid)...]`). We could think about changing this once everything is merged.

Reply:
I will merge when the tests are through. 
Then, we could merge the current master into the v0.7 branch.